homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +141,231 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
+        self.scan_coordinator = scan_coordinator
+
+        self.light = WifiLedBulb(self._ip_address)
+
+    async def _async_update_data(self):
+        """Fetch the data from this light bulb."""
+
+        await self.hass.async_add_executor_job(self.light.update_state)
+
+        get_rgbw = await self.hass.async_add_executor_job(self.light.getRgbw)
+        get_rgb = await self.hass.async_add_executor_job(self.light.getRgb)
+
+        light = {}
+
+        if self.light.mode == "ww":
+            light["mode"] = MODE_WHITE
+        elif self.light.rgbwcapable:
+            light["mode"] = MODE_RGBW
+        else:
+            light["mode"] = MODE_RGB
+
+        light["white_value"] = get_rgbw[3]
+
+        if light["mode"] == MODE_WHITE:
+            light["brightness"] = light["white_value"]
+        else:
+            light["brightness"] = self.light.brightness
+
+        light["hs_color"] = color_util.color_RGB_to_hs(*get_rgb)
+
+        light["current_effect"] = self.light.raw_state[3]
+
+        if self.light.is_on and light["brightness"] > 0:
+            light["state"] = True
+        else:
+            light["state"] = False
+
+        return light
+
+
+class FluxLight(CoordinatorEntity, LightEntity):
+    """Represents a Flux Light entity."""
+
+    def __init__(
+        self,
+        coordinator: FluxLEDCoordinator,
+        unique_id: str,
+        device: dict,
+        effect_speed: int,
+        hass: HomeAssistant,
+    ):
+        """Initialize the Flux light entity."""
+        super().__init__(coordinator=coordinator)
+
+        self._name = device[CONF_NAME]
+        self._unique_id = unique_id
+        self._icon = "mdi:lightbulb"
+        self._attrs = {}
+        self._last_update = 0
+        self._state = None
+        self._brightness = None
+        self._hs_color = None
+        self._bulb = coordinator.light
+        self._last_brightness = 255
+        self._last_hs_color = color_util.color_RGB_to_hs(255, 255, 255)
+        self._ip_address = device[CONF_HOST]
+        self._effect_speed = effect_speed
+        self._mode = coordinator.data["mode"]
+
+        async def async_remove_light(device: dict):
+            """Remove a bulb device when it is removed from options."""
+
+            bulb_id = device["device_id"]
+
+            if self._unique_id != bulb_id:
+                return
+
+            entity_registry = await hass.helpers.entity_registry.async_get_registry()
+            entity_entry = entity_registry.async_get(self.entity_id)
+
+            device_registry = await hass.helpers.device_registry.async_get_registry()
+            device_entry = device_registry.async_get(entity_entry.device_id)
+
+            if (
+                len(
+                    async_entries_for_device(
+                        entity_registry,
+                        entity_entry.device_id,
+                        include_disabled_entities=True,
+                    )
+                )
+                == 1
```

Comment:
Add a comment why it's important to check against 1.

Reply:
ok